h4l/selection/lepton.py

```diff
@@ -33,67 +34,67 @@ def electron_selection(
     pt = events.Electron.pt
     fSCeta = abs(events.Electron.eta + events.Electron.deltaEtaSC)
 
-    def return_cuts():
-
-        era = self.config_inst.campaign.x.year
-        if era == 2017 or era == 2018:
-            if self.config_inst.campaign.x.version < 10:
-                # Using 2017 WP and training (ElectronMVAEstimatorRun2Fall17IsoV2Values) since this is the only one available in Run2 UL nanoAODs. noqa
-                BDT = events.Electron.mvaFall17V2Iso
-            else:
-                BDT = events.Electron.mvaHZZIso
-
-            # pre-UL WP for Run II (miniAOD branch: Run2_CutBased_BTag16)
-            if self.config_inst.campaign.x.preUL:  
-                # print("This is preUL!")
-                # from IPython import embed; embed()
-                return (
-                    (
-                        (pt <= 10.) & (
-                            (fSCeta < 0.8 & BDT > 0.85216885148) |
-                            (fSCeta >= 0.8 & fSCeta < 1.479 & BDT > 0.82684550976) |
-                            (fSCeta >= 1.479 & BDT > 0.86937630022)
-                        )
-                    ) |
-                    (
-                        (pt > 10.) & (
-                            (fSCeta < 0.8 & BDT > 0.98248928759) |
-                            (fSCeta >= 0.8 & fSCeta < 1.479 & BDT > 0.96919224579) |
-                            (fSCeta >= 1.479 & BDT > 0.79349796445))
```

Comment:
I do think we need a function (or something similar) here. For the FSR correction, you need to apply the lepton selections twice, once at the loose working point and once at the tight. The selection criteria in this function are only needed for the tight selection, which is why I think it's good to collect them slightly separated from the other selections. Please change this back

Reply:
Hi Philip, sorry but I think there might be some misunderstanding. When we say that FSR is applied on "loose" leptons, we do not mean that we identify and isolate leptons using the loose WPs, but rather that for FSR recovery we use what the analysis refers to as "loose leptons", i.e leptons with pT > 5/7 GeV (ele/mu), |eta|<2.5, and dx,y,z cuts, w/o any ID or ISO applied.

Reply:
Also clarified in person, what I meant by "loose" WP is just not applying the ID criteria in the first place, not using the loose WP. Sorry for the confusion!